### PR TITLE
A simple cake model to simulate formation damage due to suspended solids in injection water

### PR DESCRIPTION
**Pull request by GitPaean**
downstream of https://github.com/OPM/opm-common/pull/3313

### CODE REVIEW

**Comment by GitPaean** [comment]
Will rebase this after merging the PR #4686 .

**Comment by GitPaean** [comment]
jenkins build this opm-common=3313 please

**Comment by GitPaean** [comment]
jenkins build this opm-common=3313 please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
I am marking this to be ready for review now.

**Comment by GitPaean** [comment]
jenkins build this please

File: opm/simulators/wells/WellInterfaceGeneric.cpp [hunk 1]
```diff
@@ -714,4 +718,74 @@ checkNegativeWellPotentials(std::vector<double>& well_potentials,
...
+            const double perm = filter_cake.perm;
+            const double rw = connection.getFilterCakeRadius();
+            const auto cr0 = connection.r0();
+            const auto crw = connection.rw();
+            const auto cskinfactor = connection.skinFactor();
+            const double K = connection.Kh() / connection.connectionLength();
+            const double factor = filter_cake.sf_multiplier;
+            // compute a multiplier for the transmissibility
+            if (filter_cake.geometry == FilterCake::FilterCakeGeometry::LINEAR) {
+                // but we are using this form just for first prototype
```
Comment (bska) [inline opm/simulators/wells/WellInterfaceGeneric.cpp:773]: What does this comment mean?

File: opm/simulators/wells/WellInterfaceGeneric.cpp [hunk 2]
```diff
@@ -714,4 +718,74 @@ checkNegativeWellPotentials(std::vector<double>& well_potentials,
...
+    const double conc = this->well_ecl_.getFilterConc();
+    if (conc == 0.) {
+        return;
+    }
+
+    // it is currently used for the filter cake modeling related to formation damage study
+    auto& ws = well_state.well(this->index_of_well_);
+    const auto& connection_rates = ws.perf_data.phase_rates;
+
+    const size_t np = well_state.numPhases();
```
Comment (bska) [inline opm/simulators/wells/WellInterfaceGeneric.cpp:743]: Would you mind switching to `std::size_t` from `<cstddef>`, please?

File: opm/simulators/wells/WellInterfaceGeneric.cpp (hunk 1 above)
Comment (GitPaean) [reply opm/simulators/wells/WellInterfaceGeneric.cpp:773]: It is a leftover from the cleaning up. Will be removed.

File: opm/simulators/wells/WellInterfaceGeneric.cpp (hunk 2 above)
Comment (GitPaean) [reply opm/simulators/wells/WellInterfaceGeneric.cpp:743]: will do. Thanks.

**Comment by GitPaean** [comment]
Thanks for the prompt review. The comments have been addressed and force pushed.

**Comment by GitPaean** [comment]
jenkins build this update_data please

**Comment by GitPaean** [comment]
jenkins build this opm-tests=109 please

**Comment by GitPaean** [comment]
jenkins build this opm-tests=109 please

**Comment by GitPaean** [comment]
jenkins build this opm-tests=1009 please

**Comment by GitPaean** [comment]
jenkins build this opm-tests=1009 please

**Comment by GitPaean** [comment]
I am done updating. It is appreciated that the PR can go in soon. Thanks.

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.  I'll merge into master.

**Pull request merged**